auto-pos-creator/src/multi_contract_interactions/create_pos_endpoints.rs

```diff
@@ -100,6 +103,38 @@ pub trait CreatePosEndpointsModule:
         self.create_pos_common(args)
     }
 
+    #[payable("*")]
+    #[endpoint(createPosFromLp)]
+    fn create_pos_from_lp(&self, steps: StepsToPerform) -> EsdtTokenPayment {
```

Comment:
When you call this endpoint from the microservice, you call the StepsToPerform variable as 0, 1 or 2. What happens when my input is 3 for example? Does the tx fail automatically? Or it will continue, but it will be unable to match any of the enum elements?

Reply:
Maybe we could try to have an opt_payment = match 
EnterFarm => try_enter_farm_with_lp
EnterMetastaking => try_enter_metastaking_with_lp_farm_tokens
_ => sc_panic!

After that point you only check if there is some and send it to user. But this approach only if the tx does not fail automatically.